src/types/index.ts

```diff
@@ -341,38 +346,45 @@ export interface NullableStrError {
   error: string | null;
 }
 
-export interface TextFileWatcherReadyArgs {
+export type TextChange = {
+  from: number;
+  to?: number;
+  insert?: string;
+};
+
+export type WriteChange = (changes: TextChange | Array<TextChange>) => void;
+
+export interface TextFileReadyEvent {
+  writeChange: (changes: TextChange | Array<TextChange>) => void;
+  getLatestContent: () => string;
   initialContent: string;
-  version: number;
-  writeChange: (writeChangeArgs: WriteChangeArgs) => Promise<void>;
 }
-export interface TextFileWatcherOnChangeArgs {
+export interface TextFileOnChangeEvent {
+  changes: Array<TextChange>;
   latestContent: string;
-  version: number;
-  changeSource: string;
-  changes: any; // TODO fix
-}
-export interface OnMoveOrDeleteArgs {
-  eventType: "MOVE" | "DELETE";
-  node: FsNode;
 }
-export type OnActiveFileChangeCallback = (file: string) => void;
-export type WatchFileWatcherOnChange = (newContent: string) => void;
-export type WatchFileWatcherOnError = (error: string) => void;
-export type WatchFileWatcherOnMoveOrDelete = (args: OnMoveOrDeleteArgs) => void;
-export type WatchTextFileWatcherOnReady = (
-  readyArgs: TextFileWatcherReadyArgs
+
+export type OnActiveFileChangeListener = (file: string) => void;
+export type WatchFileOnChangeListener<T extends string | Blob = string> = (
+  newContent: T
+) => void;
+export type WatchFileOnErrorListener = (error: string) => void;
+export type WatchFileOnMoveOrDeleteListener = (
+  moveOrDeleteEvent: MoveEvent | DeleteEvent
+) => void;
+export type WatchTextFileOnReadyListener = (
+  readyEvent: TextFileReadyEvent
 ) => void;
-export type WatchTextFileWatcherOnChange = (
-  changeArgs: TextFileWatcherOnChangeArgs
+export type WatchTextFileOnChangeListener = (
+  changeEvent: TextFileOnChangeEvent
 ) => void;
-export type WatchTextFileWatcherOnError = (error: string) => void;
-export type WatchTextFileWatcherOnMoveOrDelete = (
-  args: OnMoveOrDeleteArgs
+export type WatchTextFileOnErrorListener = (error: string) => void;
+export type WatchTextFileOnMoveOrDeleteListener = (
+  moveOrDeleteEvent: MoveEvent | DeleteEvent
```

Comment:
yeah all these will need update to docs. @Conner1115 is it automated or will this need manual work on your end? 

we should automate if not

Reply:
Updating docs is automated with typedoc.